web/packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.test.ts

```diff
@@ -97,14 +87,119 @@ it('removeItemsBelongingToRootCluster removes connections', () => {
     },
   ];
 
-  const service = getTestSetup({ connections });
+  const service = getTestSetupWithMockedConnections({ connections });
   service.removeItemsBelongingToRootCluster('/clusters/localhost');
   expect(service.getConnections()).toEqual([
     { clusterName: 'remote_leaf', ...connections[3] },
   ]);
 });
 
 it('updates the port of a gateway connection when the underlying doc gets updated', () => {
```

Comment:
git shows a weird diff here but basically what I've done is I extracted the common setup out of this test case into `getTestSetupWithMockedDocuments` and added two new test cases.